Enable nullable on NetConf/Scp/SshClient

mus65 opened the pull request:
Followup #1339

mus65 committed:
Enable nullable on NetConf/Scp/SshClient

mus65 commented:
CI failure looks unrelated. Windows works and I can't reproduce it locally on Linux.

mus65 committed:
Merge branch 'develop' into nullable

mus65 committed:
fix formatting

mus65 committed:
Merge branch 'develop' into nullable

Rob-Hague commented on src/Renci.SshNet/ScpClient.cs:
```diff
@@ -747,7 +784,14 @@ private void InternalDownload(IChannelSession channel, Stream input, FileSystemI
                     directoryCounter--;
 
-                    currentDirectoryFullName = new DirectoryInfo(currentDirectoryFullName).Parent.FullName;
+                    var currentDirectoryParent = new DirectoryInfo(currentDirectoryFullName).Parent;
+
+                    if (currentDirectoryParent is null)
+                    {
+                        break;
+                    }
+
+                    currentDirectoryFullName = currentDirectoryParent.FullName;
```
I think this actually fixes a NullReferenceException when downloading a directory to C:\\. It would be more explicit if the `directoryCounter == 0` check were first (I can't add as a suggestion):

```c#
directoryCounter--;

if (directoryCounter == 0)
{
    break;
}

var currentDirectoryParent = new DirectoryInfo(currentDirectoryFullName).Parent;

Debug.Assert(currentDirectoryParent is not null, $"Should be {directoryCounter} levels deeper than {startDirectoryFullName}.");

currentDirectoryFullName = currentDirectoryParent.FullName;

continue;
```

mus65 replied:
pushed. Had to add a `!` to `currentDirectoryParent!.FullName` though because legacy frameworks don't have the nullable annotation on `Debug.Assert`. :(

mus65 replied:
@Rob-Hague I just pushed another change to simply disable the nullable warnings on old frameworks since these kind of false positives would become very annoying when migrating more of the codebase to nullable.

mus65 committed:
improve directoryCounter check

mus65 committed:
Merge remote-tracking branch 'upstream/develop' into nullable

mus65 committed:
disable nullable warnings on old frameworks

since the libraries are missing a lot of nullable attributes
in the old frameworks, this causes a lot of false positive
nullable warnings.

Simply disable these warnings for old frameworks.

Rob-Hague reviewed (approved):
Thanks
